docs/frontend/40-git.rst

```diff
@@ -1,22 +1,133 @@
+
+.. todo:: More guidelines
+
+.. note::
+  On the "Conversation" tab of the PR, next to the merge-button, there are instructions on
+  how to check out and test the PR locally with the CLI.
 
 .. seealso::
-    :ref:`github-git`
\ No newline at end of file
+  https://help.github.com/articles/about-pull-request-reviews/
+  https://help.github.com/articles/reviewing-proposed-changes-in-a-pull-request/
+
+Merging a PR
+------------
+
+If you are the only, or last, reviewer of a PR, you should also merge the PR.
+
+:guilabel:`Rebase and merge` will add the commits directly on master.
+This is a good starting point and should be used for any PR only containing one commit.
+If there are more than one commit, they should make sense to have in the commit history individually.
+
+:guilabel:`Squash and merge` will squash all commits into one.
+This is a good approach for a PR that contains detail changes or a minor feature but has
+grown in number of commits, maybe due to number of requested changes during review.
+If it feels like the commits mostly reference the same thing and the individual commits would
+add more noise than traceability to the git history, this would be a good candidate.
+
+:guilabel:`Create a merge commit` will create a merge commit along with all the commits in the PR.
+This is the default git behavior and makes sense if the PR is a bigger feature with several parts,
+so they should be kept together as they affect the same feature, but they also make sense individually.
+
+Deployment and branches
+~~~~~~~~~~~~~~~~~~~~~~~
+
+
+Artifacts / binaries
+--------------------
+
+Once the build promotion pipeline (integration -> alpha -> beta -> prod) has started, we
+need a way to guarantee consistent builds across environments.
+
+.. todo:: Define a way to do this - copy integration build output files to different environments?
+
+   .. seealso:: yarn lock - https://yarnpkg.com/lang/en/docs/yarn-lock/
+
+``master`` branch
+-----------------
+
+``master`` branch will get all updates indended to reach the production environment.
+This should be done by creating a branch and a PR for code that is intended for next production release.
+
+Commits added to the master branch will build, and on success, deploy to Cosmoz integration environment.
+If integration tests succeed, the build will be promoted to the Alpha environment (https://alpha.cosmoz.com)
+for Q/A and testing.
+
+Once a release-ready state has been reached in Alpha, a tag is created for the release.
+New tags will automatically be built and deployed to Beta environment (https://beta.cosmoz.com) which 
+enables customer testing.
+
+.. todo:: Lock down ``master`` branch for direct commits ?
+
+``prod`` branch
+---------------
+
+During the next scheduled production release, unless reviewed beta-version has been rejected.
+The beta-reviewed version/tag is merged to the prod branch and pushed, which causes a build and deploy
+to Production environment (https://app.cosmoz.com).
```

Comment:
Or the prepared release for prod is deployed.

We could also secure deployments with different passwords per allowed users.